src/execute.cxx

```diff
@@ -294,9 +297,14 @@ bool ExecuteCommand::Execute(const std::string &filename)
  */
 int ExecuteCommand::Join()
 {
+    // Allow primary command to conclude
+    // ensures stdout and stderr readers
+    // exit only once primary command process
+    // has concluded
+    int commandError = this->exit_code_future.get();
     if(!this->child_out_future.get())
         return -999;
     if(!this->child_err_future.get())
         return -999;
-    return this->exit_code_future.get();
+    return commandError;
```

Comment:
(question) I'm not clear on why reordering these has an influence on process termination sequence: calling `get` doesn't trigger any terminations.